vscode/src/client/analyzerClient.ts

```diff
@@ -109,21 +110,29 @@ export class AnalyzerClient {
     }
 
     this.outputChannel.appendLine(`Starting the kai rpc server ...`);
-    this.fireStateChange("starting");
+
+    if (this.isServerRunning()) {
```

Comment:
This is kinda the same check as the TODO comment above.  Should we keep those check state TODOs around?

Reply:
Ah right yeah this probably isn't needed. Will remove.